xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -251,8 +251,11 @@ void CXBMCApp::onResume()
     RequestVisibleBehind(true);
 
   if (g_application.IsInitialized())
-    dynamic_cast<CAndroidPowerSyscall*>(CServiceBroker::GetPowerManager().GetPowerSyscall())
-        ->SetOnResume();
+  {
+    IPowerSyscall* syscall = CServiceBroker::GetPowerManager().GetPowerSyscall();
+    if (syscall)
```

Comment:
While chances are highly reduced that this now segfaults, the two commands are not atomic. Means it might go away right after the if (syscall) was checked, right?

Reply:
Right. Any ideas for improvement?

Reply:
Sadly not, only shared_ptr would help, by first assigning them, then keeping them on the stack, using them and let them go out of scope.

Reply:
Yep, that's the only solution that came to my mind.